src/mdserver/main/MDServerConnection.C

```diff
@@ -2055,7 +2058,7 @@ MDServerConnection::GetFilteredFileList(GetFileListRPC::FileList &files)
         int stage5 = visitTimer->StartTimer();
         int vfIndex = 0;
         bool needToSortFileList = false;
-        for(fileIndex = 0; fileIndex < files.names.size(); ++fileIndex)
+        for(fileIndex = 0; virtualFilesToCheck.types.size() && (fileIndex < files.names.size()); ++fileIndex)
```

Comment:
`!virtualFilesToCheck.types.empty()` might be clearer.  And faster if this is a list rather than a vector.

Reply:
Ok, so you mean this, right?

```suggestion
        for(fileIndex = 0; !virtualFilesToCheck.types.empty() && (fileIndex < files.names.size()); ++fileIndex)
```

Reply:
Yeah...tested it too.